tests/integration/components/flexberry-dropdown-test.js

```diff
@@ -0,0 +1,612 @@
+  assert.strictEqual($dropdownText.hasClass('default text'), true, 'Component\'s wrapper has \'default text\' css-class');
+  assert.strictEqual($dropdownMenu.hasClass('menu'), true, 'Component\'s wrapper has \'menu\' css-class');
+
+  // Check wrapper's additional CSS-classes.
+  let additioanlCssClasses = 'scrolling compact fluid';
+  this.set('class', additioanlCssClasses);
+  Ember.A(additioanlCssClasses.split(' ')).forEach((cssClassName, index) => {
+    assert.strictEqual(
+    $component.hasClass(cssClassName),
+    true,
+    'Component\'s wrapper has additional css class \'' + cssClassName + '\'');
+  });
+
+  // Clean up wrapper's additional CSS-classes.
+  this.set('class', '');
+  Ember.A(additioanlCssClasses.split(' ')).forEach((cssClassName, index) => {
+    assert.strictEqual(
+    $component.hasClass(cssClassName),
+    false,
+    'Component\'s wrapper hasn\'t additional css class \'' + cssClassName + '\'');
+  });
+});
+
+test('it renders i18n-ed placeholder', function(assert) {
+  assert.expect(2);
+
+  // Render component.
+  this.render(hbs`{{flexberry-dropdown
+    class=class
+  }}`);
+
+  // Retrieve component.
+  let $component = this.$().children();
+  let $dropdownText = $component.children('div.default.text');
+
+  // Check <dropdown>'s placeholder.
+  assert.strictEqual(
+    Ember.$.trim($dropdownText.text()),
+    Ember.get(I18nRuLocale, 'components.flexberry-dropdown.placeholder'),
+    'Component\'s inner <dropdown>\'s placeholder is equals to it\'s default value from i18n locales/ru/translations');
+
+  // Change current locale to 'en' & check <dropdown>'s placeholder again.
+  this.set('i18n.locale', 'en');
+  assert.strictEqual(
+    Ember.$.trim($dropdownText.text()),
+    Ember.get(I18nEnLocale, 'components.flexberry-dropdown.placeholder'),
+    'Component\'s inner <dropdown>\'s placeholder is equals to it\'s value from i18n locales/en/translations');
+});
+
+test('it renders manually defined placeholder', function(assert) {
+  assert.expect(2);
+
+  // Render component.
+  this.render(hbs`{{flexberry-dropdown
+    placeholder=placeholder
+  }}`);
+
+  // Set <dropdown>'s placeholder' & render component.
+  let placeholder = 'please type some text';
+  this.set('placeholder', placeholder);
+
+  // Retrieve component.
+  let $component = this.$().children();
+  let $dropdownText = $component.children('div.default.text');
+
+  // Check <dropdown>'s placeholder.
+  assert.strictEqual(
+    Ember.$.trim($dropdownText.text()), placeholder);
+
+  // Change placeholder's value & check <dropdown>'s placeholder again.
+  placeholder = 'dropdown has no value';
+  this.set('placeholder', placeholder);
+  assert.strictEqual(Ember.$.trim($dropdownText.text()), placeholder);
+});
+
+test('readonly mode works properly', function(assert) {
+  assert.expect(4);
+
+  // Render component.
+  this.render(hbs`{{flexberry-dropdown
+    class=class
```

Comment:
You should enable/disable readonly mode not through additional css-classs 'dicabled', but through component's 'readonly' property: readonly=readonly, .... this.set('readonly', ture);